apis/v1/httproute_types.go

```diff
@@ -196,8 +196,11 @@ type HTTPRouteRule struct {
 	// Filters define the filters that are applied to requests that match
 	// this rule.
 	//
-	// The effects of ordering of multiple behaviors are currently unspecified.
-	// This can change in the future based on feedback during the alpha stage.
+	// This spec does not mandate a specific ordering for the execution of filters.
+	// However an implementation may decide to execute the filters in the listed order.
+	// In case the implementation cannot support the listed order (due to its internal
+	// ordering limitation), the implementation will drop the route rule with such a
+	// conflict and set the status RouteConditionPartiallyInvalid for the Route.
```

Comment:
I'm not really sure how to word this, but at a high level, I think it makes sense for us to encourage all implementations to implement filters in the order they are specified wherever possible. The stricter mode also seems like it's worth mentioning as a potential option here. Note that the "SHOULD" and "MAY" are both using the [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt) definitions of those words (as elsewhere in the spec).

```suggestion
	// Wherever possible, implementations SHOULD implement filters in the order
	// they are specified. 
	//
	// Implementations MAY choose to implement this ordering strictly, rejecting
	// any combination or order of filters that can not be supported. If implementations
	// choose a strict interpretation of filter ordering, they MUST clearly document
	// that behavior.
	//
	// To reject an invalid combination or order of filters, implementations SHOULD
	// consider the Route Rule(s) with this configuration invalid. If all Route Rule(s)
	// in a Route are invalid, the entire Route would be considered invalid. In only
	// a portion of Route Rules are invalid, implementations should set the 
	// "PartiallyInvalid" condition for the Route.
```

I'm not 100% sure on this particular variation, so would appreciate some additional feedback.

/cc @arkodg @howardjohn @mlavacca @sunjayBhatia @youngnick

Reply:
I like the suggested wording by @robscott. Waiting to see the feedback from requested folks and after that I will accept it.

Reply:
I like the wording as well, although I would replace the last "should" with either MUST or SHOULD, so it's clear we're using the RFC meaning.

That is:
```
		// a portion of Route Rules are invalid, implementations (MUST|SHOULD) set the 
		// "PartiallyInvalid" condition for the Route.
```

I can't remember if the PartiallyInvalid condition is MUST or SHOULD elsewhere, sorry.

Reply:
Changed the `should` to `MUST` as suggested